promql/promqltest/test.go

```diff
@@ -376,6 +413,22 @@ func (t *test) parseEval(lines []string, i int) (int, *evalCmd, error) {
 			break
 		}
 
+		if strings.Split(defLine, " ")[0] == "expect" {
+			annoType, expectedAnno, err := parseExpect(defLine, i)
+			if err != nil {
+				return i, nil, formatErr("%w", err)
+			}
+			if cmd.expectedCmds == nil {
+				cmd.expectedCmds = make(map[expectCmdType][]*expectCmd)
```

Comment:
We generally avoid `make` if it isn't needed. This can be simply written as
```suggestion
				cmd.expectedCmds = map[expectCmdType][]*expectCmd{}
```

However, let's avoid this whole `if` by creating this map in `newInstantEvalCmd` and `newRangeEvalCmd` (similar to the `metrics` and `expected` maps).